NeuroID/src/main/java/com/neuroid/tracker/callbacks/NIDSensorHelper.kt

```diff
@@ -34,65 +34,76 @@ object NIDSensorHelper {
                     nidSensors.accelerometer.status = NIDSensorStatus.AVAILABLE
                 }
             }
-            NIDLog.i(TAG, "Sensor:${sensor.name} ${sensor.type}")
+            logger.i(TAG, "Sensor:${sensor.name} ${sensor.type}")
         }
 
-        restartSensors()
+        restartSensors(nSensors)
     }
 
     private fun initSensorManager(context: Context) {
         if (sensorManager == null)
             sensorManager = context.getSystemService(Context.SENSOR_SERVICE) as SensorManager
     }
 
+    @VisibleForTesting
+    // only called by unit tests, do not call outside of tests
+    internal fun resetSensorManager() {
+        sensorManager = null
+    }
+
+
     fun stopSensors() {
         sensorManager?.unregisterListener(listener)
     }
 
-    fun restartSensors() {
-        listener = NIDSensorGenListener {
+    fun restartSensors(nSensors: NIDSensors = nidSensors) {
+        listener = getNIDGenListener(nSensors, firstValuesGyro, firstValuesAccel)
+
+        gyroscopeSensor?.let {
```

Comment:
where is this defined/coming from? is it from the `listener` on ln 60?

Reply:
gyroscopicSensor is set in the initSensorHelper() line 27.